shopinvader_sale_coupon/tests/test_cart.py

```diff
@@ -196,27 +313,23 @@ def test_promotion_on_next_order(self):
             program.id,
             "Coupons for next order should've been generated",
         )
-        self.assertEqual(
-            generated_coupon["partner"]["id"],
-            self.partner.id,
-            "Coupons for next order should've been generated for this partner",
-        )
 
     def test_reward_amount(self):
-        self.service.dispatch(
+        cart = self.service.dispatch(
             "add_item",
             params={
                 "product_id": self.product_A.id,
                 "item_qty": 1.0,
             },
         )
-        cart = self.service.dispatch(
-            "add_item",
-            params={
-                "product_id": self.product_B.id,
-                "item_qty": 2.0,
-            },
-        )
+        # FIXME: compute_reward_total changed: in v14 all lines had a price,
+        # and for tax included products, the amount on the SO was the price
+        # tax excluded.
+        # Here in v16 the price on the line is 0 (as the product is free),
+        # and the compute goes on the product to take the lst_price.
+        # But for tax included products, the lst_price includes the tax...
+        # To sum up, reward_amount sometimes contains the tax (for tax
+        # included products) and otherwise not...
```

Comment:
TODO: 
compute_reward_total changed: in v14 all SOL had a price (even reward lines), and for tax included products, the amount on the SOL was the price
tax excluded.
Here in v16 the price on the line is 0 (as the product is free),
and the compute goes on the product to take the lst_price.
But for tax included products, the lst_price includes the tax...
To sum up, in v16 reward_amount takes care of the tax for tax
included products, and not for the other products.